async-nats/src/client.rs

```diff
@@ -45,20 +47,43 @@ impl error::Error for PublishError {}
 /// [crate::connect] and [crate::ConnectOptions::connect]
 #[derive(Clone, Debug)]
 pub struct Client {
+    info: tokio::sync::watch::Receiver<ServerInfo>,
     sender: mpsc::Sender<Command>,
     next_subscription_id: Arc<AtomicU64>,
     subscription_capacity: usize,
 }
 
 impl Client {
-    pub(crate) fn new(sender: mpsc::Sender<Command>, capacity: usize) -> Client {
+    pub(crate) fn new(
+        info: tokio::sync::watch::Receiver<ServerInfo>,
+        sender: mpsc::Sender<Command>,
+        capacity: usize,
+    ) -> Client {
         Client {
+            info,
             sender,
             next_subscription_id: Arc::new(AtomicU64::new(0)),
             subscription_capacity: capacity,
         }
     }
 
+    /// returns last received info from the server.
+    ///
+    /// Examples
+    ///
+    /// ```no_run
+    /// # #[tokio::main]
+    /// # async fn main () -> Result<(), async_nats::Error> {
+    /// let client = async_nats::connect("demo.nats.io").await?;
+    /// println!("info: {:?}", client.info());
+    /// # Ok(())
+    /// # }
+    /// ```
+    pub fn info(&self) -> ServerInfo {
```

Comment:
For stream etc, info makes sense, its "self info".
In this case, server_info? or super explicit last_known_server_info?

```suggestion
    pub fn info(&self) -> ServerInfo {
```

Reply:
hm, server_info is enough imo.

Reply:
sgtm